pkg/controllers/resources/pods/syncer.go

```diff
@@ -220,6 +277,124 @@ func (s *podSyncer) SyncDown(ctx *synccontext.SyncContext, vObj client.Object) (
 	return s.SyncDownCreate(ctx, vPod, pPod)
 }
 
+func (s *podSyncer) checkAndRewriteHostPath(ctx *synccontext.SyncContext, pPod *corev1.Pod) *corev1.Pod {
+	if len(pPod.Spec.Volumes) > 0 {
+		ctx.Log.Debugf("checking for hostpath volumes")
+
+		containersWithPhysicalMountPaths := ContainerPhysicalMountPathRegister{
+			KubeletMountPath: make(map[string]bool),
+			LogMountPath:     make(map[string]bool),
+		}
+
+		for i, volume := range pPod.Spec.Volumes {
+			if volume.HostPath != nil {
+				if volume.HostPath.Path == PodLoggingHostpathPath &&
+					// avoid recursive rewriting of HostPaths across reconciles
+					!strings.HasSuffix(volume.Name, PhysicalVolumeNameSuffix) {
+					// we can't just mount the new hostpath to the virtual log path
+					// we also need the actual 'physical' hostpath to be mounted
+					// at a separate location and added to the correct containers as
+					// only then the symlink targets created by hostpath-mapper would be
+					// able to point to the actual log files to be traced.
+					// Also we need to make sure this physical log path is not a
+					// path used by the scraping agent - which should only see the
+					// virtual log path
+					ctx.Log.Infof("rewriting hostPath for pPod %s", pPod.Name)
+					pPod.Spec.Volumes[i].HostPath.Path = s.virtualPodLogsPath
+
+					ctx.Log.Infof("adding original hostPath to relevant containers")
+					pPod = s.addPhysicalPathToVolumesAndCorrectContainers(ctx,
+						volume.Name,
+						volume.HostPath.Type,
+						PodLoggingHostpathPath,
+						PhysicalLogVolumeMountPath,
+						containersWithPhysicalMountPaths,
+						pPod)
+				}
+
+				if volume.HostPath.Path == KubeletPodPath &&
+					!strings.HasSuffix(volume.Name, PhysicalVolumeNameSuffix) {
+					ctx.Log.Infof("rewriting hostPath for kubelet pods %s", pPod.Name)
+					pPod.Spec.Volumes[i].HostPath.Path = s.virtualKubeletPodPath
+					ctx.Log.Infof("adding original hostPath to relevant containers")
+					pPod = s.addPhysicalPathToVolumesAndCorrectContainers(ctx,
+						volume.Name,
+						volume.HostPath.Type,
+						KubeletPodPath,
+						PhysicalKubeletVolumeMountPath,
+						containersWithPhysicalMountPaths,
+						pPod)
+				}
+
+				if volume.HostPath.Path == LogHostpathPath {
+					pPod.Spec.Volumes[i].HostPath.Path = s.virtualLogsPath
+				}
```

Comment:
Why is the physical volume(`PhysicalLogVolumeMountPath`) not mounted in this^ case?
I think it needs to be. Have a look at my file structure when the "filebeat" pod mounts `/var/log`:

> root@elk-filebeat-filebeat-5z6kx:/var/log/containers# ls -la
> lrwxrwxrwx. 1 root root 101 Oct  6 22:16 coredns-66ffcc6b58-tsmk2_kube-system_coredns-ea86caae31738763933394f8288230572e3266882c7e0728a9055a5faacf940d.log -> /var/log/pods/kube-system_coredns-66ffcc6b58-tsmk2_82728908-ba9d-4c0d-8212-2434313cee2a/coredns/0.log
> ...
> root@elk-filebeat-filebeat-5z6kx:/var/log/containers# ls -la /var/log/pods
> lrwxrwxrwx. 1 root root 127 Oct  6 22:16 kube-system_coredns-66ffcc6b58-tsmk2_82728908-ba9d-4c0d-8212-2434313cee2a -> /var/vcluster/physical/log/pods/vcluster_coredns-66ffcc6b58-tsmk2-x-kube-system-x-vcluster_1e71df1e-5e10-4f63-9503-d681769440b8
> ...
> root@elk-filebeat-filebeat-5z6kx:/var/log/containers# ls /var
> backups  cache  lib  local  lock  log  mail  opt  run  spool  tmp
>
> root@elk-filebeat-filebeat-5z6kx:/var/log/containers# tail coredns-66ffcc6b58-tsmk2_kube-system_coredns-ea86caae31738763933394f8288230572e3266882c7e0728a9055a5faacf940d.log 
> tail: cannot open 'coredns-66ffcc6b58-tsmk2_kube-system_coredns-ea86caae31738763933394f8288230572e3266882c7e0728a9055a5faacf940d.log' for reading: No such file or directory

Reply:
Yes I think this is because when testing against fluentd it mounts both `/var/log` and `/var/log/pods` which then won't require an explicit physical log mount but I think we need to include that. Nice catch!